From: Joel Selvaraj <foss@joelselvaraj.com>
Date: Sun, 16 Mar 2025 20:12:10 -0500
Subject: arm64: dts: qcom: sdm845-xiaomi-beryllium: add rear camera actuator

Xiaomi Poco F1 uses the ROHM BU64748 VCM as the focus lens
for its main rear camera (Sony IMX363).

Signed-off-by: Joel Selvaraj <foss@joelselvaraj.com>
---
 .../boot/dts/qcom/sdm845-xiaomi-beryllium-common.dtsi | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/sdm845-xiaomi-beryllium-common.dtsi b/arch/arm64/boot/dts/qcom/sdm845-xiaomi-beryllium-common.dtsi
index 123eb79..fa8b321 100644
--- a/arch/arm64/boot/dts/qcom/sdm845-xiaomi-beryllium-common.dtsi
+++ b/arch/arm64/boot/dts/qcom/sdm845-xiaomi-beryllium-common.dtsi
@@ -170,6 +170,16 @@ cam0_dvdd: reg_cam0_dvdd {
 		enable-active-high;
 		gpio = <&pm8998_gpios 11 GPIO_ACTIVE_HIGH>;
 	};
+
+	cam0_actuator_regulator: gpio-regulator@5 {
+		compatible = "regulator-fixed";
+		regulator-name = "cam0_actuator_regulator";
+		regulator-min-microvolt = <2800000>;
+		regulator-max-microvolt = <2800000>;
+		regulator-enable-ramp-delay = <100>;
+		enable-active-high;
+		gpio = <&tlmm 26 GPIO_ACTIVE_HIGH>;
+	};
 };
 
 &adsp_pas {
@@ -855,6 +865,14 @@ csiphy0_ep: endpoint {
 };
 
 &cci_i2c0 {
+	actuator: actuator@c {
+		compatible = "rohm,bu64748";
+		reg = <0x0c>;
+		vcc-supply = <&cam0_actuator_regulator>;
+		vio-supply = <&cam_iovdd>;
+		status = "okay";
+	};
+
 	camera@10 {
 		compatible = "sony,imx363";
 		reg = <0x10>;
@@ -871,6 +889,7 @@ camera@10 {
 		vana-supply = <&cam0_avdd>;
 		vdig-supply = <&cam0_dvdd>;
 
+		lens-focus = <&actuator>;
 		orientation = <1>;
 		rotation = <270>;
 
